code/game/objects/items/weapons/shields.dm

```diff
@@ -84,3 +83,50 @@
 		playsound(user, 'sound/weapons/saberoff.ogg', 25, TRUE)
 		to_chat(user, "<span class='notice'>[src] can now be concealed.</span>")
 	add_fingerprint(user, "turned [active ? "on" : "off"]")
+
+/obj/item/weapon/shield/ballistic
+	name = "Ballistic shield"
+	desc = "A mlitary grade ballistic shield adept at blocking bullets and melee objects from connecting with the shield wielder."
+	icon = 'icons/obj/items/weapons.dmi'
+	icon_state = "ballistic_shield"
+	max_integrity = 200
+	flags_equip_slot = ITEM_SLOT_BACK
+	force = 5
+	throwforce = 5
+	throw_speed = 1
+	throw_range = 4
+	w_class = WEIGHT_CLASS_BULKY
+	materials = list(/datum/material/metal = 1000)
+	attack_verb = list("shoved", "bashed")
+	hit_sound = 'sound/effects/grillehit.ogg'
+	destroy_sound = 'sound/effects/glassbr3.ogg'
+	var/cooldown = 0 //shield bash cooldown. based on world.time
+
+/obj/item/weapon/shield/ballistic/set_shield()
+    AddComponent(/datum/component/shield, SHIELD_PARENT_INTEGRITY,shield_cover = list("melee" = 100, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 80, "bio" = 30, "rad" = 0, "fire" = 80, "acid" = 80))
+
+
+/obj/item/weapon/shield/ballistic/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	var/health_status = (obj_integrity * 100) / max_integrity
+	switch(health_status)
+		if(0 to 20)
+			to_chat(user, "<span class='notice'>It's falling appart, will not be able to withstand much further damage.</span>")
+		if(20 to 40)
+			to_chat(user, "<span class='notice'>It has cracked edges and dents.</span>")
+		if(40 to 60)
+			to_chat(user, "<span class='notice'>It appears damaged, but still sturdy.</span>")
+		if(60 to 80)
+			to_chat(user, "<span class='notice'>It appears in decent condition, with some damage marks.</span>")
+		if(80 to 100)
+			to_chat(user, "<span class='notice'>It appears in perfect condition.</span>")
+
+/obj/item/weapon/shield/ballistic/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/baton) && world.time >= cooldown)
+		user.visible_message("<span class='warning'>[user] bashes [src] with [I]!</span>")
+		playsound(user.loc, 'sound/effects/shieldbash.ogg', 25, 1)
+		cooldown = world.time + 2.5 SECONDS
```

Comment:
We have new cooldown macros from Roh
`COOLDOWN_CHECK(src, COOLDOWN_SHIELD_BASH)` 
`COOLDOWN_START(src, COOLDOWN_SHIELD_BASH, 2.5 SECONDS)`

You'd need to define `COOLDOWN_SHIELD_BASH`
`#define COOLDOWN_SHIELD_BASH 'cooldown_shield_bash'`